src/chainstate/coordinator/mod.rs

```diff
@@ -613,7 +654,45 @@ impl<
             };
 
             // at this point, we need to figure out if the sortition we are
-            //  about to process is the first block in reward cycle.
+            //  about to process is the first block in the exit reward cycle.
```

Comment:
can this be a self-contained function to keep the function length from growing too big?

Reply:
Seconded. This should be a separate function.